mantle/cmd/kola/testiso.go

```diff
@@ -401,116 +451,61 @@ func runTestIso(cmd *cobra.Command, args []string) error {
-		if kola.CosaBuild.Meta.BuildArtifacts.LiveIso == nil {
-			return fmt.Errorf("build %s has no live ISO", kola.CosaBuild.Meta.Name)
-		}
-		ranTest = true
-		instIso := baseInst // Pretend this is Rust and I wrote .copy()
-		duration, err := testLiveIso(ctx, instIso, filepath.Join(outputDir, scenarioISOInstall), false, false)
-		printResult(scenarioISOInstall, duration, err)
-		if err != nil {
-			return err
-		}
-	}
-	if _, ok := targetScenarios[scenarioISOOfflineInstall]; ok {
-		if kola.CosaBuild.Meta.BuildArtifacts.LiveIso == nil {
-			return fmt.Errorf("build %s has no live ISO", kola.CosaBuild.Meta.Name)
-		}
-		ranTest = true
-		instIso := baseInst // Pretend this is Rust and I wrote .copy()
-		duration, err := testLiveIso(ctx, instIso, filepath.Join(outputDir, scenarioISOOfflineInstall), true, false)
-		printResult(scenarioISOOfflineInstall, duration, err)
-		if err != nil {
-			return err
+		fmt.Printf("Testing test: %s\n", test)
+		components := strings.Split(test, ".")
+
+		if kola.HasString("4k", components) {
+			enable4k = true
 		}
-	}
-	if _, ok := targetScenarios[scenarioISOLiveLogin]; ok {
-		if kola.CosaBuild.Meta.BuildArtifacts.LiveIso == nil {
-			return fmt.Errorf("build %s has no live ISO", kola.CosaBuild.Meta.Name)
+		if kola.HasString("nm", components) {
+			addNmKeyfile = true
 		}
-		ranTest = true
-		duration, err := testLiveLogin(ctx, filepath.Join(outputDir, scenarioISOLiveLogin))
-		printResult(scenarioISOLiveLogin, duration, err)
-		if err != nil {
-			return err
+		if kola.HasString("mpath", components) {
+			enableMultipath = true
 		}
-	}
-	if _, ok := targetScenarios[scenarioISOAsDisk]; ok {
-		if kola.CosaBuild.Meta.BuildArtifacts.LiveIso == nil {
-			return fmt.Errorf("build %s has no live ISO", kola.CosaBuild.Meta.Name)
+		if kola.HasString("secure", components) {
+			enableUefiSecure = true
+		} else if kola.HasString("uefi", components) {
+			 enableUefi = true
 		}
-		switch coreosarch.CurrentRpmArch() {
-		case "x86_64":
-			ranTest = true
-			duration, err := testAsDisk(ctx, filepath.Join(outputDir, scenarioISOAsDisk))
-			printResult(scenarioISOAsDisk, duration, err)
-			if err != nil {
-				return err
+
+		if kola.HasString("pxe", components) {
+			if kola.HasString("offline", components) {
+				duration, err = testPXE(ctx, inst, filepath.Join(outputDir, test), true)
+			} else {
+				duration, err = testPXE(ctx, inst, filepath.Join(outputDir, test), false)
+			}
+		} else {
+			if kola.HasString("login", components) {
+				duration, err = testLiveLogin(ctx, filepath.Join(outputDir, test))
+			} else if strings.Contains(test, "disk") {
+				duration, err = testAsDisk(ctx, filepath.Join(outputDir, test))
+			} else {
+				if kola.HasString("offline", components) {
+					duration, err = testLiveIso(ctx, inst, filepath.Join(outputDir, test), true, false)
+				} else {
+					duration, err = testLiveIso(ctx, inst, filepath.Join(outputDir, test), false, false)
+				}
```

Comment:
again the partial matches here make it harder to reason about which tests follow which code path IMO. It would be nice to get this down to a switch/case statement that we can use to easily run the different test functions based on the test name.

I took a stab at reworking this to use a switch/case statement instead. See the last two commits in https://github.com/dustymabe/coreos-assembler/commits/PR3298